src/storage/secondary/block.rs

```diff
@@ -51,6 +51,9 @@ pub trait BlockBuilder<A: Array> {
     /// Get estimated size of block. Will be useful on runlength or compression encoding.
     fn estimated_size(&self) -> usize;
 
+    /// Get distinct values count of block
+    fn distinct_count(&self) -> usize;
```

Comment:
We will have `null_count` and many more statistics in the future. Can we squash all of them into one function?

e.g. `get_statistics(&self) -> Vec<BlockStatistics>`